### PR TITLE
refactor commtrack sms handlers location

### PR DESCRIPTION
**Pull request by snopoke**
@calellowitz @biyeun 

Saw this code and had to fix it.

### CODE REVIEW

File: corehq/apps/commtrack/sms.py [hunk 1]
```diff
@@ -90,9 +91,9 @@ def __init__(self, domain, v):
                 raise NotAUserClassError
 
             # currently only support one location on the UI
-            linked_loc = u.location
-            if linked_loc:
-                self.location = get_supply_point(self.domain.name, loc=linked_loc)
+            self.location = u.location
```
Comment (snopoke) [inline corehq/apps/commtrack/sms.py]: Originally throughout this code `self.location` was a dict `{'case': supply_point, 'location': location}`. This PR basically separates that out into `self.location` and `self.case` and then tidies up the functions that fetch the location and the case.

**Comment by czue** [comment]
build fail seems real (circular imports?)

```
Traceback (most recent call last):
  File "manage.py", line 73, in <module>
    execute_from_command_line(sys.argv)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute
    django.setup()
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/__init__.py", line 21, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/django/apps/config.py", line 87, in create
    module = import_module(entry)
  File "/opt/python/2.7.9/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "custom/_legacy/a5288/__init__.py", line 1, in <module>
    from a5288.reports import MissedCallbackReport
  File "custom/_legacy/a5288/reports.py", line 5, in <module>
    from corehq.apps.reports.standard import CustomProjectReport
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/reports/standard/__init__.py", line 7, in <module>
    from corehq.apps.groups.models import Group
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/groups/models.py", line 15, in <module>
    from corehq.apps.locations.models import SQLLocation
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/locations/models.py", line 20, in <module>
    from .dbaccessors import get_all_users_by_location
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/locations/dbaccessors.py", line 3, in <module>
    from corehq.apps.locations.models import Location
ImportError: cannot import name Location
```

File: corehq/apps/commtrack/sms.py (hunk 1 above)
Comment (czue) [inline corehq/apps/commtrack/sms.py]: just trying to keep you guessing

File: corehq/apps/locations/dbaccessors.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
 from itertools import imap
+
+from corehq.apps.locations.models import Location
```
Comment (czue) [inline corehq/apps/locations/dbaccessors.py]: guessing this is the import that needs to move inline

File: corehq/apps/commtrack/util.py [hunk 3]
```diff
@@ -33,20 +37,18 @@ def all_sms_codes(domain):
...
 
-def get_supply_point(domain, site_code=None, loc=None):
-    if loc is None:
-        loc = Location.view('commtrack/locations_by_code',
-                            key=[domain, site_code.lower()],
-                            include_docs=True).first()
-    if loc:
-        case = SupplyPointCase.get_by_location(loc)
+def get_supply_point_and_location(domain, site_code):
+    location = Location.view(
```
Comment (czue) [inline corehq/apps/commtrack/util.py]: should this be calling `get_location_from_site_code`?

**Comment by czue** [comment]
> Saw this code and had to fix it.

:bear:

**Comment by snopoke** [comment]
Will address the tests once https://github.com/dimagi/commcare-hq/pull/9138 is merged.

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**